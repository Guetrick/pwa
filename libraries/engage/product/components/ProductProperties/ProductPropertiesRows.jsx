import React from 'react';
import PropTypes from 'prop-types';
import { ProductPropertiesRowUngrouped } from './ProductPropertiesRowUngrouped';
import { ProductPropertiesSubGroups } from './ProductPropertiesSubGroups';

/**
 * @param {Object} props The component props.
 * @return {JSX}
 */
export function ProductPropertiesRows({ properties }) {
  return (
    <React.Fragment>
      <ProductPropertiesRowUngrouped properties={properties} />
      <ProductPropertiesSubGroups properties={properties} />
    </React.Fragment>
  );
}

ProductPropertiesRows.propTypes = {
  properties: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};
