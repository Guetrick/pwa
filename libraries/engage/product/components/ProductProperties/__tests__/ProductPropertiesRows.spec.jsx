import React from 'react';
import { shallow } from 'enzyme';
import { ProductPropertiesRows } from '../ProductPropertiesRows';

const properties = [
  {
    label: 'test1',
    value: '123',
  },
  {
    label: 'test2',
    value: '456',
  },
  {
    label: 'test3',
    value: '789',
    subDisplayGroup: 'TestGroup',
  },
];

describe('<ProductPropertiesRows />', () => {
  it('should not render if properties are empty', () => {
    const wrapper = shallow(<ProductPropertiesRows properties={[]} />);
    expect(wrapper.instance()).toEqual(null);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render grouped and ungrouped properties', () => {
    const wrapper = shallow(<ProductPropertiesRows properties={properties} />);
    expect(wrapper.find('ProductPropertiesRowUngrouped').length).toEqual(1);
    expect(wrapper.find('ProductPropertiesSubGroups').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
