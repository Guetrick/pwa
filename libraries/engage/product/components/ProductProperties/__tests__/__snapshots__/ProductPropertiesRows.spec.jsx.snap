// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPropertiesRows /> should not render if properties are empty 1`] = `
<Fragment>
  <ProductPropertiesRowUngrouped
    properties={Array []}
  />
  <ProductPropertiesSubGroups
    properties={Array []}
  />
</Fragment>
`;

exports[`<ProductPropertiesRows /> should render grouped and ungrouped properties 1`] = `
<Fragment>
  <ProductPropertiesRowUngrouped
    properties={
      Array [
        Object {
          "label": "test1",
          "value": "123",
        },
        Object {
          "label": "test2",
          "value": "456",
        },
        Object {
          "label": "test3",
          "subDisplayGroup": "TestGroup",
          "value": "789",
        },
      ]
    }
  />
  <ProductPropertiesSubGroups
    properties={
      Array [
        Object {
          "label": "test1",
          "value": "123",
        },
        Object {
          "label": "test2",
          "value": "456",
        },
        Object {
          "label": "test3",
          "subDisplayGroup": "TestGroup",
          "value": "789",
        },
      ]
    }
  />
</Fragment>
`;
